toolchain/check/handle_name.cpp

```diff
@@ -185,7 +184,7 @@ auto HandleMemberAccessExpr(Context& context, Parse::MemberAccessExprId node_id)
             << " of function type";
         if (IsInstanceMethod(context.sem_ir(), function_decl->function_id)) {
           context.AddInstAndPush(
-              {node_id,
+              {Parse::MemberAccessExprId{node_id},
```

Comment:
Do we know that the `node_id` is definitely a `MemberAccessExprId` here? I think it could be a `PointerMemberAccessExprId` instead. You'll probably need to change the definition of `BoundMethod` to allow either kind of `NodeId` -- use `Parse::NodeIdOneOf` -- and change the type of `node_id` in this function to match. (A typedef for `AnyMemberAccessExprId` might be useful.)